### ISSUE TITLE
Support passing options to transforms

### ISSUE THREAD

**Issue by devinivy** [devinivy/loveboat#1]
Transforms should be able to take options, similarly in form to the way hapi plugins may receive options.

Those options should be able to be passed alongside a transform any place that it's registered.  The options should then be passed to the transform's handler.  Additionally, I think we should support extra `before`s and `after`s this way.

**Comment by devinivy** [comment]
Example,

``` js
// Currently
server.routeTransforms([MyTransform]);

// We want
server.routeTransforms([
  {
    transform: MyTransform,
    options: { something: 'dogs' },
    before: ['other-transform'],
    after: ['other-other-transform']
  }
]);
```

**Comment by devinivy** [comment]
Resolved by #8
